src/coreComponents/constitutive/docs/CO2Brine.rst

```diff
@@ -1,80 +1,256 @@
 .. _CO2-EOS:
 
 ##################################################################################
-CO2 and brine models 
+CO2-brine model 
 ##################################################################################
 
 
 Summary
-=================================
+=======
 
-The equation-of-state and viscosity of CO2 under both sub- and super-critical conditions are computed as a function of both pressure and temperature, using the empirical equations developed by Span & Wagner (1996) and Fenghour & Wakeman (1998), respectively.
+The CO2-brine model implemented in GEOSX includes two components (CO2 and H2O) that are transported by one or two fluid phases (the gas (:math:`g`) phase and the brine (:math:`\ell`) phase).
+The water component is only present in the brine phase, while the CO2 component can be present in the gas phase as well as in the brine phase.
+Thus, considering the phase component fractions, :math:`y_{c,p}` (i.e., the fraction of the mass of phase :math:`p` represented by component :math:`c`) the following partition matrix determines the component distribution within the two phases:
 
-The brine density, which depends on pressure, temperature, and salinity, is calculated based on the correlation developed by Phillips et al. (1981).
+.. math::
+    \begin{bmatrix}
+    y_{CO2,g} & y_{CO2,\ell} \\
+         0 & 1            \\
+    \end{bmatrix}
 
+The update of the fluid properties is done in two steps:
 
-Function Description
-=================================
+1) The phase fractions (:math:`\nu_p`) and phase component fractions (:math:`y_{c,p}`) are computed as a function of pressure (:math:`p`), temperature (:math:`T`), component fractions (:math:`z_c`), and a constant salinity.
 
-**Function of CO2 density calculation**
+2) The phase densities (:math:`\rho_p`) and phase viscosities (:math:`\mu_p`) are computed as a function of pressure, temperature, the updated phase component fractions, and a constant salinity.
 
- .. code-block:: c
+Once the phase fractions, phase component fractions, phase densities, phase viscosities--and their derivatives with respect to pressure, temperature, and component fractions--have been computed, the :ref:`CompositionalMultiphaseFlow` proceeds to the assembly of the accumulation and flux terms.
+Note that the current implementation of the flow solver is isothermal and that the derivatives with respect to temperature are therefore discarded.
```

Comment:
@CusiniM should we leave this note here, or will it be outdated soon?  Maybe revise to: "If the solver is isothermal, the derivates w.r.t. temp are discarded"?

Reply:
I think we can leave it for now and I will update it in the thermalCompositionalSolver PR so that the docs gets updated along with the code.